packages/tupaia-web/src/features/Dashboard/Dashboard.tsx

```diff
@@ -103,17 +104,34 @@ const DashboardImageContainer = styled.div`
 `;
 
 export const Dashboard = () => {
+  const navigate = useNavigate();
   const { projectCode, entityCode, dashboardName } = useParams();
+  const { data: project, isLoading: isLoadingProject } = useProject(projectCode);
   const { dashboards, activeDashboard } = useDashboards(projectCode, entityCode, dashboardName);
   const [isExpanded, setIsExpanded] = useState(false);
   const [exportModalOpen, setExportModalOpen] = useState(false);
   const { data: entity } = useEntity(projectCode, entityCode);
   const bounds = entity?.bounds || DEFAULT_BOUNDS;
+  const defaultEntityLink = useEntityLink(entityCode);
 
   const toggleExpanded = () => {
     setIsExpanded(!isExpanded);
   };
 
+  // check for valid dashboard name, and if not valid and not still loading, redirect to default dashboard
+  useEffect(() => {
+    // if is valid or loading, don't redirect
+    if (
+      !dashboards ||
+      isLoadingProject ||
+      project?.code !== projectCode ||
+      dashboards?.find(dashboard => dashboard.name === dashboardName)
+    )
+      return;
+
+    navigate(defaultEntityLink);
+  }, [JSON.stringify(dashboards), dashboardName, projectCode, isLoadingProject]);
```

Comment:
I was on the fence about whether to put this here or as a redirect in the `Routes` file, but went with here to match the pattern with default map overlays in the Map file